dask/dataframe/io/parquet/fastparquet.py

```diff
@@ -288,154 +278,408 @@ def read_metadata(
+                            tz = getattr(dtypes[col], "tz", None)
+                            cmin = pd.Timestamp(cmin, tz=tz)
+                            cmax = pd.Timestamp(cmax, tz=tz)
+                        cnull = column.meta_data.statistics.null_count
+                        last = cmax_last.get(name, None)
+
+                        if not filters:
+                            # Only think about bailing if we don't need
+                            # stats for filtering
+                            if cmin is None or (last and cmin < last):
+                                # We are collecting statistics for divisions
+                                # only (no filters) - Column isn't sorted, or
+                                # we have an all-null partition, so lets bail.
+                                #
+                                # Note: This assumes ascending order.
+                                #
+                                gather_statistics = False
+                                file_row_group_stats = {}
+                                file_row_group_column_stats = {}
+                                break
+
+                        if single_rg_parts:
+                            s["columns"].append(
+                                {
+                                    "name": name,
+                                    "min": cmin,
+                                    "max": cmax,
+                                    "null_count": cnull,
+                                }
+                            )
+                        else:
+                            cstats += [cmin, cmax, cnull]
+                        cmax_last[name] = cmax
+                    else:
+                        if not filters and column.meta_data.num_values > 0:
+                            # We are collecting statistics for divisions
+                            # only (no filters) - Lets bail.
+                            gather_statistics = False
+                            file_row_group_stats = {}
+                            file_row_group_column_stats = {}
+                            break
+
+                        if single_rg_parts:
+                            s["columns"].append({"name": name})
+                        else:
+                            cstats += [None, None, None]
+                if gather_statistics:
+                    file_row_group_stats[fpath].append(s)
+                    if not single_rg_parts:
+                        file_row_group_column_stats[fpath].append(tuple(cstats))
+
+        return (
+            file_row_groups,
+            file_row_group_stats,
+            file_row_group_column_stats,
+            gather_statistics,
+            global_lookup,
+            base_path,
+        )
+
+    @classmethod
+    def _get_thrift_row_groups(
+        cls,
+        pf,
+        filename,
+        row_groups,
+        global_lookup,
+    ):
+        real_row_groups = []
```

Comment:
Comment/docstring: turn a set of row-groups into bytes-serialised form using thrift via pickle.